x/observer/keeper/msg_server_add_block_header.go

```diff
@@ -35,6 +34,38 @@ func (k msgServer) AddBlockHeader(goCtx context.Context, msg *types.MsgAddBlockH
 		return nil, cosmoserrors.Wrapf(types.ErrBlockHeaderVerficationDisabled, "proof verification not enabled for evm ,chain id: %d", msg.ChainId)
 	}
 
+	_, found = k.GetBlockHeader(ctx, msg.BlockHash)
+	if found {
```

Comment:
What if we have a bug and a wrong header is added?
Shouldn't we consider the capability to update an existing header?

Reply:
I was also thinking about the capability of replacing a existing block header (in store), but seems it happens only when protocol has a bug or when majority zetaclients (or EVM node providers) get corrupted, which should be sufficiently rare.